File: app/src-tauri/src/utils/paths.rs
use std::{env, path::{Path, PathBuf}};

pub fn project_root() -> PathBuf {
    let cwd = env::current_dir().unwrap_or_else(|_| PathBuf::from("."));
    if cwd.ends_with(Path::new(r"app\src-tauri")) || cwd.ends_with(Path::new("app/src-tauri")) {
        cwd.parent()
            .and_then(|p| p.parent())
            .map(Path::to_path_buf)
            .unwrap_or(cwd)
    } else if cwd.ends_with(Path::new("app")) {
        cwd.parent().map(Path::to_path_buf).unwrap_or(cwd)
    } else {
        cwd
    }
}

pub fn resolve_from_project_root(value: &str) -> PathBuf {
    let path = PathBuf::from(value);
    if path.is_absolute() {
        path
    } else {
        project_root().join(path)
    }
}

File: app/electron/services/paths.ts
import { existsSync } from 'node:fs'
import path from 'node:path'

export const projectRoot = () => {
  const cwd = process.cwd()
  if (cwd.endsWith('app')) {
    return path.dirname(cwd)
  }
  if (existsSync(path.join(cwd, 'app', 'package.json'))) {
    return cwd
  }
  return path.dirname(cwd)
}

export const appRoot = () => path.join(projectRoot(), 'app')
export const resolveFromProjectRoot = (value: string) => (path.isAbsolute(value) ? value : path.join(projectRoot(), value))
export const profilesRoot = () => path.join(projectRoot(), 'data', 'profiles')
export const logsRoot = () => path.join(projectRoot(), 'data', 'logs')
export const extensionsRoot = () => path.join(projectRoot(), 'data', 'extensions')
export const runtimeRoot = () => path.join(projectRoot(), 'runtime')
export const automationScriptsDir = () => path.join(projectRoot(), 'scripts', 'automation')
export const toDisplayPath = (value: string) => path.relative(projectRoot(), value).replace(/\\/g, '/') || value.replace(/\\/g, '/')
